src/session/sessions.ts

```diff
@@ -0,0 +1,284 @@
+//#region > Imports
+//#PACKAGE "graphql"
+//## npm install "graphql"@14.6.0
+// Contains the interface for gql queries, mutations and subscriptions
+import { DocumentNode } from "graphql";
+//#PACKAGE "js-cookie"
+//## npm install "js-cookie"@2.2.1
+// A simple, lightweight JavaScript API for handling browser cookies
+import Cookies from "js-cookie";
+
+//> Session
+// Contains the base session
+import Session from "./index";
+//> Templates
+// Contains the main template for the sessions
+import { IMainTemplate } from "../templates/index";
+// Contains the SNEK template
+import SnekTemplate from "../templates/snek/index";
+//> Tasks
+// Contains SNEK tasks
+import SnekTasks from "../templates/snek/gql/tasks/index";
+//> Interfaces
+// Contains the interface for the apollo endpoint
+import { ApolloEndpoint } from "../../src/endpoints/index";
+// Contains basic session interfaces
+import { User } from "./index";
+//#endregion
+
+//#region > Classes
+/** @class A Github SubSession */
+class GithubSession extends Session {
+  /**
+   * @constructor
+   * @author Nico Schett <contact@schett.net>
+   * @param {string} sId A session name
+   * @param {Endpoint} ep A endpoint
+   * @param {IMainTemplate} template A template set
+   */
+  constructor(
+    sId: string,
+    public ep: ApolloEndpoint,
+    public template: IMainTemplate
+  ) {
+    /** @todo Change template set to dedicated github template set */
+    super(sId);
+  }
+
+  /**
+   * Send a query to the endpoint.
+   *
+   * @param {string} token A authentication token
+   * @param {DocumentNode} data A DocumentNode with a query
+   * @param {object} variables A object with variables
+   */
+  async send(token: string, data: DocumentNode, variables?: object) {
+    let headers = {
+      authorization: token,
+    };
+
+    return this.ep.sendQuery(data, variables, headers);
+  }
+}
+
+/** @class A SNEK SubSession */
+class SnekSession extends Session {
+  refreshTokenName: string = "refresh";
+  /* Define tasks */
+  public tasks: SnekTasks;
+
+  /**
+   * @constructor
+   * @author Nico Schett <contact@schett.net>
+   * @param {string} sId A session name
+   * @param {Endpoint} ep A endpoint
+   * @param {SnekTemplate} template A template set
+   */
```

Comment:
Should have some sort of comment.

Initializes a SNEK session.

Reply:
Yeah, but not your suggestion cause this includes too less/useless information.
We have to discuss about a new TONIC rule in snek-at/tonic/issues/3